client/src/components/SearchResultsOffset.js

```diff
@@ -0,0 +1,166 @@
+import React from 'react'
+import { Box, Button, Paragraph, TextInput } from 'grommet'
+import { FormPrevious, FormNext } from 'grommet-icons'
+import { isOnlyNumbers } from '../helpers/isOnlyNumbers'
+import { useSearchResources } from '../hooks/useSearchResources'
+
+export const SearchResultsOffset = () => {
+  const {
+    query: { limit, offset },
+    response: { count },
+    setOffset,
+    goToSearchResults
+  } = useSearchResources()
+
+  const [enteredPageNumber, setEnteredPageNumber] = React.useState('')
+  const [
+    enteredPageNumberInRange,
+    setEnteredPageNumberInRange
+  ] = React.useState(false)
+
+  const safeOffset = parseInt(offset, 10) || 0
+  const safeCount = parseInt(count, 10) || 0
+  const parsedLimit = parseInt(limit, 10)
+  const lastOffset =
+    safeCount && parsedLimit > 0 ? Math.floor(safeCount / parsedLimit) : 0
+
+  const startingPageSlices =
+    safeOffset <= 2 ? [0, 6] : [safeOffset - 2, safeOffset + 3]
+
+  const pageSlices = lastOffset - safeOffset <= 2 ? [-5] : startingPageSlices
+
+  const pageButtonsOffsets = [...Array(lastOffset + 1).keys()].slice(
+    ...pageSlices
+  )
+
+  const goToOffset = (page) => {
+    setOffset(page)
+    setEnteredPageNumber('')
+    goToSearchResults()
+  }
+
+  const atEnd = safeOffset === lastOffset
+  const atStart = safeOffset === 0
+
+  const handlePageNumberRequest = ({ target: { value } }) => {
+    if (isOnlyNumbers(value)) {
+      const pageOffset = parseInt(value, 10) - 1
+      const inRange = pageOffset <= lastOffset && pageOffset >= 0
+      setEnteredPageNumberInRange(inRange)
+      setEnteredPageNumber(value)
+    }
+    if (value === '') setEnteredPageNumber('')
+  }
+
+  const goToOffsetRequest = () => {
+    goToOffset(parseInt(enteredPageNumber, 10) - 1)
```

Comment:
Why don't you check `isOnlyNumbers` here as well?

Reply:
we only need to make sure that enteredPageNumber is vetted before we set it

Reply:
Ah I see, so it's already been set so it's already been vetted, but we still have to parse it because the value has to be a string since it's a text box.

Reply:
Yeah we technically dont have to parse it i'm pretty sure it would work but here the enteredPageNumber should always be a string of numbers and the form should manage if/when you can submit it